src/org/etools/j1939tools/modules/GhgActiveTechnologyArrayModule.java

```diff
@@ -140,7 +141,7 @@ private String writeLine(ActiveTechnology active, ActiveTechnology stored, Activ
     }
 
     private String writeTechnology(ActiveTechnology technology) {
-        if (technology != null) {
+        if (technology != null && technology.getTimeSpn().hasValue() && technology.getDistanceSpn().hasValue()) {
```

Comment:
It's possible the time has a value but distance doesn't.  I'd recommend handling each separately.